fpmsyncd/routesync.cpp

```diff
@@ -758,6 +773,98 @@ void RouteSync::onRouteMsg(int nlmsg_type, struct nl_object *obj, char *vrf)
 }
 
 /* 
+ * Handle label route
+ * @arg nlmsg_type      Netlink message type
+ * @arg obj             Netlink object
+ */
+void RouteSync::onLabelRouteMsg(int nlmsg_type, struct nl_object *obj)
+{
+    struct rtnl_route *route_obj = (struct rtnl_route *)obj;
+    struct nl_addr *daddr;
+    char destaddr[MAX_ADDR_SIZE + 1] = {0};
+
+    daddr = rtnl_route_get_dst(route_obj);
+    nl_addr2str(daddr, destaddr, MAX_ADDR_SIZE);
+    SWSS_LOG_INFO("Receive new LabelRoute message dest addr: %s", destaddr);
+    if (nl_addr_iszero(daddr)) return;
+
+    if (nlmsg_type == RTM_DELROUTE)
+    {
+        m_label_routeTable.del(destaddr);
+        return;
+    }
+    else if (nlmsg_type != RTM_NEWROUTE)
+    {
+        SWSS_LOG_INFO("Unknown message-type: %d for LabelRoute %s", nlmsg_type, destaddr);
+        return;
+    }
+
+    /* Get the index of the master device */
+    uint32_t master_index = rtnl_route_get_table(route_obj);
+    /* if the table_id is not set in the route obj then route is for default vrf. */
+    if (master_index)
```

Comment:
How did you find out non-zero value implies a VRF table. "254" is the main table that ip route installs, 255 and 253 and local and default route table. All these are non-VRF tables.

Reply:
@smahesh You do know that SONiC has modified the fpm netlink message content from frr to something non-standard, right?  The rtnl_route_get_table() function will not return the RT_TABLE value in SONiC fpmsyncd, it returns the VRF ifindex value instead because SONiC has patched this.  
https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-frr/patch/0003-Use-vrf_id-for-vrf-not-tabled_id.patch

Reply:
that's for the info, no I wasn't aware.

Reply:
@smaheshm Yes, this is the reason I have to disable VRF support in the other PR (https://github.com/Azure/sonic-swss/pull/1765) where I provide the fpmsyncd runtime option to listen to kernel netlink instead of frr fpm.  kernel netlink messages send the standard RT_TABLE ID in the netlink message, whereas frr fpm substitutes the patched VRF ifindex value.